book/src/prerequisites.md

```diff
@@ -1,21 +1,73 @@
 
 Let us begin by installing all necessary tools.
-First follow the instructions on the [GTK website](https://www.gtk.org/docs/installations/) to install GTK 4.
+First follow the instructions on the [GTK website](https://www.gtk.org/docs/installations/) in order to install GTK 4.
 Then install Rust with [rustup](https://rustup.rs/).
 
 Now create a new project by executing:
 ```bash
 cargo new my-gtk-app
 ```
 
-Add the following lines to your `Cargo.toml` and you are ready to go!
+Add the following lines to your `Cargo.toml`.
 ```toml
 [dependencies.gtk]
 version = "0.1"
 package = "gtk4"
 ```
 
-# Buildsystem
+Now you can run your application by executing:
+```bash
+cargo run
+```
+
+## Cargo + Meson
+
+Cargo is *nearly* enough, but it is not well suited for handling resources such as icons or UI files.
+That is why we recommend to use [Meson](https://mesonbuild.com/) on top of it.
+It is cross-platform, its syntax is very readable and it offers many convenience functionalities for GTK.
+
+Here as well, you first follow the instructions on the [GTK website](https://www.gtk.org/docs/installations/) in order to install GTK 4.
+Then install Rust with [rustup](https://rustup.rs/).
+Finally, install Meson by following the instructions on the [Meson website](https://mesonbuild.com/Getting-meson.html).
+
+You can download a ready-to-use gtk-rust-template [here](https://gitlab.gnome.org/bilelmoussaoui/gtk-rust-template).
+Follow the instructions in the README to initialize your own application.
+Then configure your project.
+```bash
+meson --prefix=/usr build
+```
+
+In order to compile and install it run the following commands.
+You have to execute it every time you modify your application.
+```bash
+ninja -C build && ninja -C build install
+```
+
+Now the application should be in a folder included in your system path.
+You can either start it with the application launcher of your choice or in the terminal.
+
+## Cargo + Meson + Flatpak
+
+If you develop on Linux, using Flatpak is the most convenient option.
+With Flatpak your whole workflow is containerized and your users get the very same application you develop on (including all dependencies). 
+You only have to assure that Flatpak is installed on your system, check this [website](https://flatpak.org/setup/) to see if any steps are necessary on your distribution.
+Also download the [gtk-rust-template](https://gitlab.gnome.org/bilelmoussaoui/gtk-rust-template) and initialize it.
+
+Then either install:
+
+- [GNOME Builder](https://flathub.org/apps/details/org.gnome.Builder) or
+- [VSCodium](https://flathub.org/apps/details/com.vscodium.codium) together with the [rust-analyzer](https://open-vsx.org/extension/matklad/rust-analyzer) and [flatpak](https://open-vsx.org/extension/bilelmoussaoui/flatpak-vscode) extensions.
+
+That is it.
+Both IDEs will install all necessary dependencies for you, with GNOME Builder you only have to press the run button for that.
```

Comment:
That's not actually true, only Builder is capable of doing so. Unless you meant the application dependencies

Reply:
I thought runtime is installed by the extension or is this only planned?

Reply:
It was never the case? planned, yes, no idea when i will get the time to look at it. Also, currently, only the GNOME 40 runtime contains gtk4, so you can probably just be explicit about how to install both the rust sdk & gnome one. and we can remove that once both IDEs can handle installing the extensions just fine.